ads/google/a4a/utils.js

```diff
@@ -510,3 +482,48 @@ export function mergeExperimentIds(newIds, currentIdString) {
   return currentIdString + (currentIdString && newIdString ? ',' : '')
       + newIdString;
 }
+
+/**
+ * Adds two CSI signals to the given amp-analytics configuration object, one
+ * for render-start, and one for ini-load.
+ *
+ * @param {!Window} win
+ * @param {!Element} element The ad slot.
+ * @param {!JSONType} config The original config object.
+ * @param {?../../../src/service/xhr-impl.FetchResponseHeaders} responseHeaders
+ *   XHR service FetchResponseHeaders object containing the response
+ *   headers.
+ * @param {number} deltaTime The time difference, in ms, between the lifecycle
+ *   reporter's initialization and now.
+ * @param {number} initTime The initialization time, in ms, of the lifecycle
+ *   reporter.
+ *   TODO(levitzky) Remove the above two params once AV numbers stabilize.
+ * @return {?JSONType} config or null if invalid/missing.
+ */
+export function addCsiSignalsToAmpAnalyticsConfig(win, element, config,
+    responseHeaders, isVerifiedAmpCreative, deltaTime, initTime) {
+  // Add CSI pingbacks.
+  const correlator = getCorrelator(win);
+  const slotId = element.getAttribute('data-amp-slot-index');
```

Comment:
For all the attributes that you're getting with a .getAttribute call and then assigning to the baseCsiUrl, are these all set by us? I.e. should we be doing a regex test to make sure that they meet a specified format before we add them into the URL?